dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Empty.java

```diff
@@ -13,20 +11,20 @@ public Empty(Relation rel) {
         super(rel);
     }
 
-    @Override
-    public TupleSet getEncodeTupleSet(){
-        return rel.getMaxTupleSet();
-    }
+	@Override
+	public TupleSet getEncodeTupleSet(){
+		return new TupleSet();
+	}
 
-    @Override
-    public BooleanFormula consistent(SolverContext ctx) {
-    	BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
-		BooleanFormula enc = bmgr.makeTrue();
-        for(Tuple tuple : rel.getEncodeTupleSet()){
-            enc = bmgr.and(enc, bmgr.not(rel.getSMTVar(tuple, ctx)));
-        }
-        return enc;
-    }
+	@Override
+	public TupleSet getDisabledSet() {
+		return new TupleSet(rel.getMaxTupleSet());
+	}
+
+	@Override
+	public BooleanFormula consistent(SolverContext ctx) {
+		return ctx.getFormulaManager().getBooleanFormulaManager().makeTrue();
```

Comment:
Yes, as I said in some other comment, as soon as the `Empty` axiom propagates its disabled set, it becomes essentially obsolete as its complete semantic is given by its disabled set.

Reply:
This, `getEncodeTupleSet()` and their siblings should be merged with their old implementation, if we want `GlobalSettings.COMPUTE_MUSTNOT_SETS` to work.

Reply:
We need to talk about what an `must-not` set is actually supposed to be. Is it an obligation that says _"the following tuples need to be false!"_? If so, then we should have a strict subset relation with the `active set` because it entails (1) that the tuples are active (i.e. get encoded) and (2) that they shall be false.
This is not dual to `must sets` because those say _"the following tuples are true but whether you care about their value is up to you"_ (which says that `must sets` have no immediate relationship with `active sets`).

And I think this view is correct. `Active sets` are an obligation to encode values because someone depends on them, while `must-not sets` (or `disabled sets`) are also such an obligation with the additional constraint of making sure the tuples are `false`. 
In that sense, I think `Empty` should have `getEncodeTupleSet() == getDisabledSet()` and conceptually its consistency predicate is that all tuples in `encodeSet \ disabledSet` are `false`. In the case where we do not compute disabled sets, we get the original encoding we had. In the case we do, it reduces to the formula `true`.